### PR TITLE
Extended language service

### PR DESCRIPTION
**Pull request by danieltre23**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Previous commit added the extended template diagnostics integration with the compiler, but the language service uses a different function for non-ts files. Currently the language service would not generate extended diagnostics for the non-ts files.

Issue Number: #42966 

## What is the new behavior?
The language service can now generate extended template diagnostics for non-ts files.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Other information
Ignore first 5 commit, since they are in a separate PR #43107

### CODE REVIEW

File: packages/compiler-cli/src/ngtsc/typecheck/testing/index.ts [hunk 1]
```diff
@@ -20,9 +20,8 @@ import {ClassDeclaration, isNamedClassDeclaration, TypeScriptReflectionHost} fro
 import {ComponentScopeReader, LocalModuleScope, ScopeData, TypeCheckScopeRegistry} from '../../scope';
 import {makeProgram} from '../../testing';
 import {getRootDirs} from '../../util/src/typescript';
-import {ProgramTypeCheckAdapter, TemplateTypeChecker, TypeCheckContext} from '../api';
+import {ProgramTypeCheckAdapter, TemplateDiagnostic, TemplateTypeChecker, TypeCheckContext} from '../api';
 import {TemplateId, TemplateSourceMapping, TypeCheckableDirectiveMeta, TypeCheckBlockMetadata, TypeCheckingConfig} from '../api/api';
-import {TemplateDiagnostic} from '../diagnostics';
```
Comment (dgp1130) [inline packages/compiler-cli/src/ngtsc/typecheck/testing/index.ts:25]: Nit: I think we can also remove this dependency from the associated build file. It's good to do that to keep individual targets as small as possible where we can.

**Review by dgp1130 — APPROVED** [review]
Looks great! I don't really see any issues at all with this. It's particularly helpful that the first commit is just a pure move and makes this much easier to review as a result.

**Review by atscott — CHANGES_REQUESTED** [review]
Could you also add a test for inline templates? Otherwise this change LGTM

**Comment by danieltre23** [comment]
> Could you also add a test for inline templates? Otherwise this change LGTM

I actually added that test in #43107 because it worked with the compiler integration. But I can move it here if you'd prefer that.

https://github.com/angular/angular/pull/43107/files#:~:text=%7D)%3B-,it(%27should%20produce%20invalid%20banana%20in%20box%20warning%27%2C%20()%20%3D%3E%20%7B,-const%20files%20%3D%20%7B

**Comment by atscott** [comment]
Oh, I see. That's alright, as long as we have coverage somewhere for it.

**Review by atscott — APPROVED** [review]

File: packages/compiler-cli/src/ngtsc/typecheck/src/checker.ts [hunk 2]
```diff
@@ -245,6 +247,14 @@ export class TemplateTypeCheckerImpl implements TemplateTypeChecker {
           diag => convertDiagnostic(diag, fileRecord.sourceManager)));
       diagnostics.push(...shimRecord.genesisDiagnostics);
 
+      if (extendedTemplateDiagnsotics === true) {
```
Comment (dgp1130) [inline packages/compiler-cli/src/ngtsc/typecheck/src/checker.ts:250]: Nit: `if (extendedTemplateDiagnostics) { /* ... */ }`.

**Review by dgp1130 — APPROVED** [review]

**Comment by danieltre23** [comment]
Addressed some comments from #42984, in particular added the `__ngCode` brand that Alex suggested, PTAL.

**Review by alxhub — APPROVED** [review]

**Pull request closed without merging**